### PR TITLE
feat(v2): add icon to external navbar links

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

For more consistency, we should probably add icon for external links in the navbar too. This was previously done for sidebar links (see #4261).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Preview

image

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/NavbarItem/DefaultNavbarItem.tsx [hunk 1]
```diff
@@ -32,6 +35,8 @@ function NavLink({
   const toUrl = useBaseUrl(to);
   const activeBaseUrl = useBaseUrl(activeBasePath);
   const normalizedHref = useBaseUrl(href, {forcePrependBaseUrl: true});
+  const isExternalLink = label && href && href !== '#';
```
Comment (lex111) [inline packages/docusaurus-theme-classic/src/theme/NavbarItem/DefaultNavbarItem.tsx:38]: Curious, if we should count `#` as internal link? If so, we can use the built-in `isInternalUrl` function instead this condition.

**Review by slorber — COMMENTED** [review]
image

Not sure all users will appreciate having an external link icon in their navbar, some sites have a lot of external links in navbar and having this icon multiple times in navbar does not look too good imho. + I don't think there's a clear API to disable this apart swizzling.

What about limiting this for dropdown items only and not top-level navbar items? (at least for desktop?)

**Comment by lex111** [comment]
I was inspired by VuePress as example that adds icon for external links to the navbar. Actually, end user can remove this icon via CSS attribute selector.

I suggest adding this icon without explicit customization for now, and just wait for feedback from users. As for me, it's not a big change, so I do not think that there will be a lot of dissatisfied with it.

**Comment by slorber** [comment]
Hmm why not.

I think for now they can also disable it by using `to` instead of `href` (even if I don't like that "API" much)

Should we keep the release item in our own site? We don't seem to have enough width to make it fit:

image

**Review by slorber — APPROVED** [review]

**Comment by lex111** [comment]
Yes, will remove this link, it was added just for demo purposes.

However, in Infima I previously opened [PR that reduces current padding between navbar items](https://github.com/facebookincubator/infima/pull/137). This seems to make sense to me, can you check this out?

**Pull request merged**